web/src/message_fetch.ts

```diff
@@ -119,7 +119,7 @@ export function fetch_more_if_required_for_current_msg_list(
     looking_for_old_msgs: boolean,
 ): void {
     assert(message_lists.current !== undefined);
-    if (has_found_oldest && has_found_newest) {
+    if (has_found_oldest && has_found_newest && message_lists.current.visibly_empty()) {
         // Even after loading more messages, we have
         // no messages to display in this narrow.
         narrow_banner.show_empty_narrow_message();
```

Comment:
Should the other calls just below in this block also be wrapped by this check? It seems a bit strange that the code would have ever evolved to be this way. Did `show_empty_narrow_message` used to check if it's correct to show a banner or something?

Reply:
Oh, I see, the original caller for this function in this file had checked `visibly_empty` already, but this newer caller can happen without having checked that:

```
                // We don't need to fetch messages if we have enough context or                         
                // complete message history since `scroll_finished`                                     
                // automatically fetch messages if user                                                 
                // is near the end of rendered messages.                                                
                if (!has_complete_history && !has_enough_context) {                                     
                    message_fetch.fetch_more_if_required_for_current_msg_list(                          
                        has_found_oldest,                                                               
                        has_found_newest,                                                               
                        true,                                                                           
                        true,                                                                           
                    );                                                                                  
                }                                                                                       
```

Reply:
> Should the other calls just below in this block also be wrapped by this check?

No, since we want to request the server again if the first fetch resulted in only muted messages.

Reply:
No, I mean just `compose_closed_ui.update_buttons_for_private` and `compose_recipient.check_posting_policy_for_compose_box`.

I'm going to merge this but please make sure we make an explicit decision on if that's correct.

Reply:
Yeah, we need to explicitly call those. Also, this will preserve previous behaviour of updating compose in empty feed which had no inclination of being broken before.